sympy/solvers/solveset.py

```diff
@@ -125,11 +126,62 @@ def _invert_real(f, g_ys, symbol):
                             symbol)
 
     if isinstance(f, Abs):
-        pos = Interval(0, S.Infinity)
-        neg = Interval(S.NegativeInfinity, 0)
-        return _invert_real(f.args[0],
-                    Union(imageset(Lambda(n, n), g_ys).intersect(pos),
-                          imageset(Lambda(n, -n), g_ys).intersect(neg)), symbol)
+        conditions = S.EmptySet
+        condition_interval = Interval(0, oo)
+
+        if f.is_positive:
+            f_n = f
+        else:
```

Comment:
> Abs is always nonnegative so there is no need for an else, is there?

I did this because being non-negative doesn't imply a positive value.
And the `else` condition takes care of that case.

> Perhaps you meant to check the argument and are trying to make things simpler in terms of what is reported as having to be positive?

Yes. Thats exactly what I am trying to do here.